proj/src/main/webapp/script.js

```diff
@@ -139,6 +139,10 @@ if(btnSignUp){
             var allTags = {};
             for(var ii = 0; ii < tagList.length; ii++){
 
+                while(tagList[ii].charAt(ii) == ' '){
+                    tagList[ii] = tagList[ii].substr(1);
+                }
+
```

Comment:
This looks like it is trying to remove leading spaces or maybe trailing spaces? Using `charAt(ii)` looks like a mistake, though because `ii` is the index of the tag we are looking at.

Also this should be done a couple lines above this where `tagList[ii].trim()` is called.